miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/util/form/PaneAjaxTag.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2002-2014 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.ac.bbsrc.tgac.miso.webapp.util.form;
+
+import org.springframework.web.servlet.tags.RequestContextAwareTag;
+
+@SuppressWarnings("serial")
+public class PaneAjaxTag extends RequestContextAwareTag {
```

Comment:
Suggested improvement:

This class is a near-duplicate to `SearchPaneTag`.  It would probably be a good idea to make an abstract `PaneTag` class, and then have `PaneAjaxTag`, `SearchPaneTag`, and `PaneTag` extend this class, providing an appropriate JavaScript handler to create the pane (eg. `Pane.createInstrumentStatusPane`).

I haven't tried this yet, but it's probably the right thing to do, if possible.

Reply:
There should *not* be one of these per item type. You can create a JS method called `PaneTarget.xxx.createPane` and have `Pane.createPane` call that.

Reply:
This makes sense; the Tag objects should not have to pass in *both* a `PaneTarget` and a JavaScript function name.  :+1: